policy-test/tests/outbound_api.rs

```diff
@@ -623,6 +647,251 @@ async fn opaque_service() {
     .await;
 }
 
+#[tokio::test(flavor = "current_thread")]
+async fn route_with_filters() {
+    with_temp_ns(|client, ns| async move {
+        // Create a service
+        let svc = create_service(&client, &ns, "my-svc", 4191).await;
+
+        let mut rx = retry_watch_outbound_policy(&client, &ns, &svc).await;
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an initial config");
+        tracing::trace!(?config);
+
+        // There should be a default route.
+        detect_http_routes(&config, |routes| {
+            let route = assert_singleton(routes);
+            assert_route_is_default(route, &svc, 4191);
+        });
+
+        let backend_name = "backend";
+        let backends = [backend_name];
+        let _route = create(
+            &client,
+            mk_http_route(
+                &ns,
+                "foo-route",
+                &svc,
+                4191,
+                Some(&backends),
+                None,
+                Some(vec![
+                    k8s::policy::httproute::HttpRouteFilter::RequestHeaderModifier {
+                        request_header_modifier: k8s_gateway_api::HttpRequestHeaderFilter {
+                            set: Some(vec![k8s_gateway_api::HttpHeader {
+                                name: "set".to_string(),
+                                value: "set-value".to_string(),
+                            }]),
+                            add: Some(vec![k8s_gateway_api::HttpHeader {
+                                name: "add".to_string(),
+                                value: "add-value".to_string(),
+                            }]),
+                            remove: Some(vec!["remove".to_string()]),
+                        },
+                    },
+                    k8s::policy::httproute::HttpRouteFilter::RequestRedirect {
+                        request_redirect: k8s_gateway_api::HttpRequestRedirectFilter {
+                            scheme: Some("http".to_string()),
+                            hostname: Some("host".to_string()),
+                            path: Some(k8s_gateway_api::HttpPathModifier::ReplacePrefixMatch {
+                                replace_prefix_match: "/path".to_string(),
+                            }),
+                            port: Some(5555),
+                            status_code: Some(302),
+                        },
+                    },
```

Comment:
nit, take it or leave it: we could maybe have a function that returns these filters, since they're used in both tests?